lib/json.rs

```diff
@@ -45,32 +45,46 @@ pub fn to_json_value<T: serde::Serialize>(x: T) -> ddlog_std::Result<JsonValue,
     res2std(serde_json::to_value(x).map(JsonValue::from))
 }
 
-pub struct ValueWrapper(serde_json::value::Value);
-
-impl serde::Serialize for ValueWrapper {
+impl serde::Serialize for JsonValue {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: serde::Serializer,
     {
         if serializer.is_human_readable() {
-            self.0.serialize(serializer)
+            match self {
+                JsonValue::JsonNull => serializer.serialize_unit(),
+                JsonValue::JsonBool { b } => serializer.serialize_bool(*b),
+                JsonValue::JsonNumber { ref n } => val_from_num(n.clone()).serialize(serializer),
+                JsonValue::JsonString { ref s } => serializer.serialize_str(s),
+                JsonValue::JsonArray { ref a } => a.serialize(serializer),
+                JsonValue::JsonObject { ref o } => {
+                    use serde::ser::SerializeMap;
+                    let mut map = serializer.serialize_map(Some(o.len()))?;
+                    for ddlog_std::tuple2(k, v) in o.iter() {
+                        map.serialize_entry(&k, &v)?;
+                    }
+                    map.end()
+                }
+            }
         } else {
-            serde_json::to_string(&self.0)
+            serde_json::to_string(self)
                 .map_err(|e| serde::ser::Error::custom(e))?
                 .serialize(serializer)
```

Comment:
This looks really wrong, we shouldn't need this branch at all since the first branch just serializes the value and this branch will only end up calling the first branch when it serializes `self`

Reply:
That's correct, it will use the first branch to serialize itself into a JSON string and then store the string inside the serializer.